R/zchunk_batch_transportation_UCD_xml.R

```diff
@@ -8,85 +8,156 @@
+    L254.GlobalTranTechSCurve <- get_data(all_data, "L254.GlobalTranTechSCurve")
+    L254.StubTranTechCalInput <- get_data(all_data, "L254.StubTranTechCalInput")
+    L254.StubTranTechLoadFactor <- get_data(all_data, "L254.StubTranTechLoadFactor")
+    L254.StubTranTechCost <- get_data(all_data, "L254.StubTranTechCost")
+    L254.StubTranTechCoef <- get_data(all_data, "L254.StubTranTechCoef")
+    L254.StubTechCalInput_passthru <- get_data(all_data, "L254.StubTechCalInput_passthru")
+    L254.StubTechProd_nonmotor <- get_data(all_data, "L254.StubTechProd_nonmotor")
+    L254.PerCapitaBased_trn <- get_data(all_data, "L254.PerCapitaBased_trn")
+    L254.PriceElasticity_trn <- get_data(all_data, "L254.PriceElasticity_trn")
+    L254.IncomeElasticity_trn <- get_data(all_data, "L254.IncomeElasticity_trn")
+    L254.BaseService_trn <- get_data(all_data, "L254.BaseService_trn")
 
     # ===================================================
 
     # Produce outputs
-    # ...AARON - similarly, need to change these to use inputs above
-    create_xml("transportation_agg.xml") %>%
-      add_xml_data(L252.FinalEnergyKeyword_trn,"FinalEnergyKeyword") %>%
-      add_xml_data(L252.SubsectorLogit_trn,"SubsectorLogit") %>%
-      add_xml_data(L252.SubsectorShrwt_trn,"SubsectorShrwt") %>%
-      add_xml_data(L252.SubsectorInterp_trn,"SubsectorInterp") %>%
-      add_xml_data(L252.SubsectorInterpTo_trn,"SubsectorInterpTo") %>%
-      add_xml_data(L252.StubTech_trn,"StubTech") %>%
-      add_xml_data(L252.GlobalTechShrwt_trn,"GlobalTechShrwt") %>%
-      add_xml_data(L252.GlobalTechEff_trn,"GlobalTechEff") %>%
-      add_xml_data(L252.GlobalTechCost_trn,"GlobalTechCost") %>%
-      add_xml_data(L252.StubTechCalInput_trn,"StubTechCalInput") %>%
-      add_xml_data(L252.PerCapitaBased_trn,"PerCapitaBased") %>%
-      add_xml_data(L252.PriceElasticity_trn,"PriceElasticity") %>%
-      add_xml_data(L252.BaseService_trn,"BaseService") %>%
-      add_precursors("L252.FinalEnergyKeyword_trn", "L252.SubsectorLogit_trn", "L252.SubsectorShrwt_trn", "L252.SubsectorInterp_trn", "L252.SubsectorInterpTo_trn", "L252.StubTech_trn", "L252.GlobalTechShrwt_trn", "L252.GlobalTechEff_trn", "L252.GlobalTechCost_trn", "L252.StubTechCalInput_trn", "L252.PerCapitaBased_trn", "L252.PriceElasticity_trn", "L252.BaseService_trn") ->
-      transportation_agg.xml
+    create_xml(paste0("transportation_agg_", energy.TRN_SSP, ".xml")) %>%
+      add_xml_data(L254.Supplysector_trn, "Supplysector") %>%
+      add_xml_data(L254.FinalEnergyKeyword_trn, "FinalEnergyKeyword") %>%
+      add_xml_data(L254.tranSubsectorLogit, "tranSubsectorLogit") %>%
+      add_xml_data(L254.tranSubsectorSpeed, "tranSubsectorSpeed") %>%
+      add_xml_data(L254.tranSubsectorSpeed_passthru, "tranSubsectorSpeed_passthru") %>%
+      add_xml_data(L254.tranSubsectorSpeed_noVOTT, "tranSubsectorSpeed_noVOTT") %>%
+      add_xml_data(L254.tranSubsectorSpeed_nonmotor, "tranSubsectorSpeed_nonmotor") %>%
```

Comment:
It looks like in the old data system this was using the header `tranSubsectorSpeed`:
```R
write_mi_data( L254.tranSubsectorSpeed_passthru, "tranSubsectorSpeed", "ENERGY_LEVEL2_DATA", "L254.tranSubsectorSpeed_passthru", "ENERGY_XML_BATCH", paste0( "batch_transportation_UCD_", trn_SSP, ".xml" ) )
write_mi_data( L254.tranSubsectorSpeed_noVOTT, "tranSubsectorSpeed", "ENERGY_LEVEL2_DATA", "L254.tranSubsectorSpeed_noVOTT", "ENERGY_XML_BATCH", paste0( "batch_transportation_UCD_", trn_SSP, ".xml" ) )
write_mi_data( L254.tranSubsectorSpeed_nonmotor, "tranSubsectorSpeed", "ENERGY_LEVEL2_DATA", "L254.tranSubsectorSpeed_nonmotor", "ENERGY_XML_BATCH", paste0( "batch_transportation_UCD_", trn_SSP, ".xml" ) )
```

I didn't look through all of these but we should double check the `header` arguments here match the ones used in the old data system.
Note if you are still getting failures with regards to column orderings you may just need to pass `NULL` for the last argument and ensure the columns orders match manually before making the call to `add_xml_data`.